src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -427,6 +428,18 @@ public bool HasUnsavedChanges
         }
 
         /// <summary>
+        /// Returns if current workspace is readonly.
+        /// </summary>
+        public bool IsReadOnly
```

Comment:
No reason here but this implementation is open for future expansion if we want to have some kind of runtime event handling.